bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLoggerAllocator.java

```diff
@@ -203,6 +205,15 @@ private void setLastLogId(File dir, long logId) throws IOException {
     void stop() {
         // wait until the preallocation finished.
         allocatorExecutor.shutdown();
+        try {
+            if (!allocatorExecutor.awaitTermination(3, TimeUnit.SECONDS)) {
+                log.warn("Timedout while awaiting for allocatorExecutor's termination, so force shuttingdown");
+            }
+        } catch (InterruptedException e) {
+            log.warn("Got InterruptedException while awaiting termination of allocatorExecutor, so force shuttingdown");
+        }
+        allocatorExecutor.shutdownNow();
```

Comment:
fun fact: shutdownNow() does not wait for termination, so awaitTermination() is still needed after.
So why don't we just do shutdownNow() on line 207 and wait after? 
there is nothing in that executor but prealloc tasks and we don't really need them to run at this point.

Reply:
shutdownNow will *interrupt* all of the running threads. This will cause unpredictable logs/errors.

This way is cleaner